Issue와 PR 템플릿 추가

yoouyeon opened the pull request:
## 📝 관련 이슈

- close #3 

## 💻 작업 내용

- PR 템플릿 추가
- Issue 템플릿 추가
  - **`Feature`** 템플릿 : 기능 추가, 디자인 추가 등 기본 작업 전에 생성하는 이슈를 위한 템플릿
  - **`Bug`** 템플릿 : 이것은 논의했던 것에는 없긴 한데 기본 템플릿에 있기도 하고 매번 필요했던 것 같아서 추가해봤습니다.
- PR / Issue 분류용 라벨 추가
  - FE 라벨
  - 분류 라벨 (Feature, Bug, ...)
  - 우선순위 라벨 (1순위, 2순위, ...)

## 📸 스크린샷

> PR 템플릿은 실제 적용되는 것을 지금 바로 테스트해보기 어려워서... 후속 이슈에서 테스트해보려고 합니다.

| 설명 | 이미지 |
|---|---|
| Issue 템플릿 선택 화면 | <img width="678" alt="Screen_Shot 2025-01-18 00 39 29" src="https://github.com/user-attachments/assets/9e27ac4d-9fef-4940-b4c5-7c54daf4cb64" /> |
| Feature 템플릿 적용 화면 | <img width="1091" alt="Screen_Shot 2025-01-17 20 39 18" src="https://github.com/user-attachments/assets/6d5842d2-bc31-4243-8520-3241153bd0a8" /> | 
| Bug 템플릿 적용 화면 | <img width="1117" alt="Screen_Shot 2025-01-18 00 39 36" src="https://github.com/user-attachments/assets/37d2d28e-d4f9-42d5-bbe5-b0de6dc78e46" /> | 
| 라벨 목록 | <img width="989" alt="Screen_Shot 2025-01-18 01 27 23" src="https://github.com/user-attachments/assets/29049907-30f2-439c-bc35-7c82afec9e48" /> |


## 👻 리뷰 요구사항

### FE 라벨 추가

저희가 Github 프로젝트를 BE랑 함께 쓰다 보니까 이슈를 한눈에 구분할 수 있는 라벨이 있으면 좋겠다는 생각이 들더라고요. 그래서 `🖥️ FE` 라벨을 이 레포지토리에서 생성하는 모든 이슈나 PR에 붙이도록 하면 필터링하기 좋을 것 같은데 어떨까요??
(만약에 `🖥️ FE` 라벨을 사용하게 된다면 Issue에는 지금 기본적으로 붙도록 설정해두었고 PR은 나중에 스크립트로 붙이도록 추가할 예정입니다...)

### Bug 템플릿

제가 임의로 추가한 템플릿이라 추가할 내용이나 불필요한 내용이 있는지 확인 부탁드려요 🥹

## 📌 참고 사항

#2 에서 이 PR에 이어서

- PR에 라벨 붙이기
- PR Assignee 설정하기, 리뷰 요청하기

Github Action 들을 추가해보려고 합니다!

yoouyeon committed:
docs: PR 템플릿 추가

yoouyeon committed:
docs: issue 템플릿 추가

- 기능 추가 템플릿
- 버그 제보 템플릿

ongheong reviewed (commented):
PR, Issue 템플릿과 라벨까지 추가하시느라 정말 고생하셨습니다!! 🙌  
특히 추가해주신 Bug 템플릿은 버그 발생 환경이 추가된 부분이 정말 좋네요! QA시에 유용할 것 같습니다!

**리뷰 요구사항 답변**

*   FE 라벨 추가 좋습니다! 확실히 이슈를 역할별로 쉽게 구분할 수 있겠네요. 그런데 PR의 경우에는 레포가 나뉘어 있어서 어떤 경우에 필요할지 잘 모르겠습니다.
*   버그 템플릿의 `재현 방법` 파트를 좀 더 구체적으로 작성하면 좋을 것 같습니다! 
    *   다음 링크를 참고했습니다 : [[Git] GitHub Issue 사용하여 협업하기](https://mynamesieun.github.io/git/GitHub-Issue-%EC%82%AC%EC%9A%A9%ED%95%98%EC%97%AC-%ED%98%91%EC%97%85%ED%95%98%EA%B8%B0/#21-%EB%B2%84%EA%B7%B8-%EB%A6%AC%ED%8F%AC%ED%8A%B8-%EC%9D%B4%EC%8A%88-%ED%85%9C%ED%94%8C%EB%A6%BF)

```
## 📝 버그 발생 상황

<!--- (가능하면) Given-When-Then 형식으로 서술해주세요 -->

- **Given**: (문제 발생 조건 설명)
- **When**: (사용자의 행동 설명)
- **Then**: (버그가 발생한 결과 설명)

## 🎯 예상 결과

<!--- 예상했던 정상적인 결과가 어떤 것이었는지 설명해주세요 -->
```

yoouyeon committed:
docs: Bug 템플릿 내용 추가

- 재현 방법 항목 구체화 → 발생 상황 / 예상 결과
- https://github.com/dnd-side-project/dnd-12th-7-frontend/pull/4#pullrequestreview-2560346303

ongheong reviewed (approved):
버그 템플릿 추가된 내용 확인했습니다! 주말에 고생 많으셨습니다 🥹🥹 바로 머지하셔도 됩니다!

yoouyeon commented:
@ongheong 코멘트 적고 있었는데ㅎㅎㅎ 확인 감사합니다 😊
PR에도 FE 태그를 붙이려고 한것은 Project에 PR도 추가할 수 있기 때문이었는데요. 아직 이슈 관리용으로 Zenhub를 쓰는 것만 정하고 아직 구체적으로 어떻게 쓸지 얘기해본건 아니라 일단은 이슈만 추가해두고 PR에는 굳이 FE 태그를 안 붙여도 될 것 같아요!

---

아참 그리고
저희 Merge 방식은 기본 방식인 머지 커밋을 생성하는 것으로 하기로 정했었는데요.
설정 페이지 보다 보니까 General 설정에서 머지 커밋을 설정할 수 있더라구요!

<img width="333" alt="Screen_Shot 2025-01-18 19 24 54" src="https://github.com/user-attachments/assets/0fb10976-7080-43d8-b404-554b74cbe163" />

---

| Default message | <img width="680" alt="Screen_Shot 2025-01-18 19 25 53" src="https://github.com/user-attachments/assets/b7f53bc8-9ce8-493e-baee-765a19e67226" /> |
|---|---|
| Pull request title | <img width="680" alt="Screen_Shot 2025-01-18 19 25 38" src="https://github.com/user-attachments/assets/f1cb5df8-a8f3-4f2d-9eef-7f9f52eceb08" /> |

개인적으론 기본 메시지가 너무 길고 못생겨서(...) 후자인 `Pull request title` 로 설정하는 것도 괜찮을 것 같은데 어떻게 생각하시나요?? 
기존 형식과 통일하려면 `Pull request title` 앞에 `merge: ` 요렇게 붙여주는 방법도 있을 것 같구요! (일일이 붙이는게 귀찮다면 github action으로 커밋 메시지 앞에 알아서 붙도록 할 수 있을 것 같긴 해요.)

ongheong commented:
*   오 Project에서 PR을 사용하는지 몰랐네요! 그럼 Zenhub를 사용하기 전까지는 우선 해당 태그는 빼고 올리겠습니다!
*   머지 커밋의 형식을 지정할 수가 있었군요 😮 PR Title 앞에 merge를 설정하는 방법이 좋아보입니다!

yoouyeon commented:
네엡!🫡 pr title로 머지 커밋 형식 변경해두었습니다!!
